### PR TITLE
feat: Update my account page for volunteers

### PR DESCRIPTION
**Pull request by AmandaMisjuwar**
## Brief description. What is this change? 
Updated My Account page so that UI is changed for volunteers and volunteers can update their info

## Implementation description. How did you make this change?
* UI changes
* Only update donor info if user's role is donor

## Steps to test
1. Create an account/log in as a volunteer.
2. Go to My Account page and edit first name, last name, or phone number.
3. When you refresh the page, or log out and log in, the fields should be updated.

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] The appropriate tests if necessary have been written

### CODE REVIEW

**Review by hanlinc27 — APPROVED** [review]

**Review by Shehryar21 — DISMISSED** [review]
I am not able to save changes and edit volunteer account information. That is because in our useEffect function we call `getDonorByUserId()` but there is no donor and it returns an error, so some changes will need to done there so that its not called when role is Volunteer (should be simple).

**Comment by AmandaMisjuwar** [comment]
@Shehryar21 Fixed your comment. Could you also test for a donor account and make sure we can edit a field w/o error?

File: frontend/src/components/pages/Account.tsx [hunk 1]
```diff
@@ -75,9 +75,11 @@ const Account = (): JSX.Element => {
         );
       }
       setDonor(donorResponse);
-      setBusinessName(donorResponse.businessName);
+      if (authenticatedUser?.role !== Role.VOLUNTEER)
+        setBusinessName(donorResponse.businessName);
+      else setBusinessName("");
```
Comment (hanlinc27) [inline frontend/src/components/pages/Account.tsx:80]:
```suggestion
   setBusinessName(authenticatedUser?.role !== Role.VOLUNTEER ? donorResponse.businessName: "");
```

File: frontend/src/components/pages/Account.tsx [hunk 2]
```diff
@@ -75,9 +75,11 @@ const Account = (): JSX.Element => {
...
     };
-    getDonor();
+    if (authenticatedUser?.role !== Role.VOLUNTEER) getDonor();
```
Comment (hanlinc27) [inline frontend/src/components/pages/Account.tsx:82]: can we lift the `getDonor` function outside of the `useEffect`?

Comment (hanlinc27) [reply frontend/src/components/pages/Account.tsx:82]: also maybe rename to something more reflective of the function now that we have multiple roles?

**Review by hanlinc27 — APPROVED** [review]
Tested:
✅ donor can edit account info
✅ volunteer can edit account info

**Pull request merged**